python/src/aiconfig/schema.py

```diff
@@ -664,6 +668,22 @@ def update_prompt(self, prompt_name: str, prompt_data: Prompt):
                 self.prompt_index[prompt_data.name] = prompt_data
                 break
 
+    def _strip_model_names_in_prompt_data(self, prompt_data: Prompt):
+        """
+        Strips model names from the prompt data
+        """
+        if (prompt_data.metadata is not None and prompt_data.metadata.model is not None):
+            model = prompt_data.metadata.model
+            if isinstance(model, str):
+                model = model.strip()
+            else:
+                model.name = model.name.strip()
+
+                if model.settings is not None and "model" in model.settings:
+                    # TODO: Support cases where "model" field under model settings isn't just a str, since it technically can be anything depending on the model parser
+                    if isinstance(model.settings["model"], str):
+                        model.settings["model"] = model.settings["model"].strip()
```

Comment:
Was thinking on this some more. Let's not do this specific change. My reasoning is:
- parser/schema level should has no idea what this 'model' means. It's an implementation detail of the parser
- similarly, it's the parser's responsibility to use the 'model' from the settings as the 'model' for the inference run (that logic is in the parser, not the schema/core sdk)

So, if we want to strip the settings model it should be done when pulling it from the settings in the relevant parsers